### PR TITLE
fix automatic naming of migrations

### PR DESCRIPTION
**Pull request by YuriiYurchyk**
fix automatic naming of migrations when editing them in the versions section of the plugin.

Problem: 
when updating tables using the plugin in the Database section, the names of new migration files are named as follows 
'builder_table_update_users_3.php'
but when editing migrations in the Versions section they are renamed like this 
'builder_table_update_users3.php'
This can lead to naming conflicts and different naming styles for migrations

in other words
Used function Str::snake() converts the string as follows:
BuilderTableUpdateUsers3 -> builder_table_update_users3

the method I proposed converts the current
BuilderTableUpdateUsers3 -> builder_table_update_users_3

code of the new method 'toSnakeCase' is taken from here and changed
https://github.com/laravel/framework/blob/5.5/src/Illuminate/Support/Str.php#L442

When translating into English, google translite is used)

### CODE REVIEW

**Comment by YuriiYurchyk** [comment]
To reproduce the error, you need
1. Create a new table
To start numbering migrations of table updates, the following two steps are required
2. Add one column, save and apply in the DB section
3. Add one more column, save and apply in the DB section
4. Go to the version section and roll back and run the last migration
5. Add one column, save and apply in the DB section
6. Go to the version section and roll back and run the last migration

**Comment by bennothommo** [comment]
@YuriiYurchyk sorry, but we cannot accept this PR as is.

If someone uses a migration name like `Build2021UsersTable` and expects the number to stick together, given that it is a year, your regex will convert it to `build_2_0_2_1_users_table`. Given numbers have multiple meanings, there's no easy way to determine if a number is a suffix, or a year, or a version, etc. - hence why we don't try and separate them.

You will need to manually rename the migrations as you create them, if you are intending to keep a specific format.

**Comment by bennothommo** [comment]
@YuriiYurchyk I realised you are instead trying to point out an inconsistency between the two sections and how they name the files - we'll have to take a look into that.

**Comment by YuriiYurchyk** [comment]
> If someone uses a migration name like `Build2021UsersTable` and expects the number to stick together, given that it is a year, your regex will convert it to `build_2_0_2_1_users_table`.
> 
Fixed this flaw in the solution and re-filled the commit

File: classes/MigrationModel.php [hunk 1]
```diff
@@ -498,4 +497,17 @@ protected function getPluginVersionInformation()
...
     }
+
+    protected function makeScriptFileName($value)
+    {
+        $value = Str::snake($value);
+
+        $value = preg_replace_callback('/[0-9]+$/u', function ($match) {
+            $numericSuffix = $match[0];
+
+            return '_'.$numericSuffix;
```
Comment (bennothommo) [inline classes/MigrationModel.php:508]:
```suggestion
            return '_' . $numericSuffix;
```

**Comment by 7dir** [comment]
+1
